runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -973,65 +1302,317 @@ MM_SchedulingDelegate::calculateAutomaticGMPIntermission(MM_EnvironmentVLHGC *en
+	const double xmaxpct = _extensions->dnssExpectedRatioMaximum._valueSpecified * 100;
+	const double targetPauseTimeMs = (double)_extensions->tarokTargetMaxPauseTime;
+	const double pauseTimeTooHighOverheadLogBase = 1.0156;
+
+	double overhead = 0.0;
+
+	if (heapFullyExpanded) {
+		/* 
+		 * Eden size is being driven by heuristic which is trying to MINIMIZE hybrid overead, while trying to stay within tarokTargetMaxPauseTime,
+		 * The overhead logic here will map a low avg pgc time (ie, under targetPauseTimeMs), to a low overhead value (aka, a "better"/more desirable value)
+		 * Ex. Suppose tarokTargetMaxPauseTime == 100ms. 20ms -> 5% (good/desirable), 100ms -> 5% (still good), 110ms -> 6% (should possibly shrink) 500ms -> 80% (bad/undesirable/eden should probably shrink)
+		 */
+		double midpointPct = (xmaxpct + xminpct)/2.0;
+		if (pgcPauseTimeMs <= targetPauseTimeMs) {
+			/* Once the pgc time is at, or below the max pgc time, there is no "benefit" from shrinking it further, since we are already satisfying tarokTargetMaxPauseTime */
+			overhead = midpointPct;
+		} else {
+			/* 
+			 * If pgc time is above the max pgc time, map high PGC time values as very very high overhead, in efforts to bring the PGC time down to tarokTargetMaxPauseTime
+			 * If pgc time is only slightly above tarokTargetMaxPauseTime, then there is only a very small overhead penalty, 
+			 * wheras being 2x higher than the target pause time leads to a significantly bigger penalty.
+			 */
+			double overheadCurve = pow(pauseTimeTooHighOverheadLogBase, ((double)pgcPauseTimeMs - targetPauseTimeMs)) + midpointPct - 1;
+			overhead = OMR_MIN(100.0, overheadCurve);
+		}
+		
+	} else {
+		/* 
+		 * Eden sizing logic is trying to keep hybrid overhead between xminpct and xmaxpct, while trying to respect targetPauseTimeMs. 
+		 * The function/model used below adheres to the following high level concepts:
+		 * - if pgcPauseTimeMs is greater than targetPauseTimeMs, explicitly suggest contraction. ie, return a value < xminpct. Note: it is possible for this overhead to be negative if pgcPauseTimeMs is far beyond pause target - this is okay.
+		 * - if pgcPauseTimeMs < targetPauseTimeMs - 20, suggest neither explicit contraction nor explicit expansion, but rather, whatever the current cpu overhead currently is (which may be either expansion or contraction). In this case, we are far from targetPauseTimeMs, so cpu overhead can drive eden size 
+		 * - if (targetPauseTimeMs - 20) < pgcPauseTimeMs < targetPauseTimeMs, the mapped overhead is somewhere between xmaxpct and xminpct. The closer pgcPauseTimeMs is to targetPauseTimeMs, the closer the overhead will be to xminpct (which suggests contraction)
+		 * 
+		 *  NOTE: This mapped overhead value is later blended with pgc cpu overhead. This means that if the pgc pause time mapping suggested eden contraction, contraction is not guaranteed.
+		 */
+		double slope = (xminpct - xmaxpct) / 20;
```

Comment:
`20.0`?

Reply:
I'll move that to a constant at the top of the file, and add a comment describing what it means/does.

One of the models that was developed to support this mapping, uses 20, and it empirically produces good results. 
The larger this value, the smaller the "penalty" for being over the target pause time becomes smaller, whereas the closer to 0, the penalty becomes massive, which leads to PGC times respecting the pause times very very tighly.

20.0 gives a great balance. If PGC cpu overhead is suffering a lot, and can benefit from going a bit over the pause time target, then it will do so. 20 simply gives a good balance here. 

With a value like 1000 for example, the logic will barely care about the pause time target, and mostly focus on PGC overhead - This is not great, because then target pause time barely does anything.

With a value like 0.1, the pause time target will almost never be surpassed - this is NOT ideal. Certain situations will have very high PGC overhead if we fully respect a very low target pause time